lib/src/fake_cloud_firestore_instance.dart

```diff
@@ -36,9 +37,13 @@ class FakeFirebaseFirestore implements FirebaseFirestore {
       BehaviorSubject<Map<String, dynamic>?>();
   final FakeFirebaseSecurityRules securityRules;
 
-  FakeFirebaseFirestore(
-      {Stream<Map<String, dynamic>?>? authObject, String? securityRules})
-      : securityRules =
+  final FakeServerTimeProvider? fakeServerTimeProvider;
+
+  FakeFirebaseFirestore({
+    Stream<Map<String, dynamic>?>? authObject,
+    String? securityRules,
+    this.fakeServerTimeProvider,
```

Comment:
Instead of passing your own FakeServerTimeProvider, we probably want to use the clock package like the ticket reporter suggested. Clock allows us to later use FakeAsync to pass time artificially as explained at https://pub.dev/packages/clock.

Would you mind reworking the PR by removing FakeServerTimeProvider, adding clock as a dependency, then modifying the constructor like so:

```dart
  final Clock _clock;

  FakeFirebaseFirestore({
    Stream<Map<String, dynamic>?>? authObject,
    String? securityRules,
    Clock? clock,
  }) : securityRules = ..., _clock = clock ?? Clock()
```

Reply:
Sure, no problem. I will do that😃 @atn832